app/user.go

```diff
@@ -915,26 +915,81 @@ func (a *App) SetProfileImageFromFile(userId string, file io.Reader) *model.AppE
 		return model.NewAppError("SetProfileImage", "api.user.upload_profile_user.upload_profile.app_error", nil, "", http.StatusInternalServerError)
 	}
 
+	// Invalidate user cache
 	<-a.Srv.Store.User().UpdateLastPictureUpdate(userId)
+	a.invalidateUserCacheAndPublish(userId)
 
-	a.InvalidateCacheForUser(userId)
+	return nil
+}
 
-	user, userErr := a.GetUser(userId)
-	if userErr != nil {
-		mlog.Error(fmt.Sprintf("Error in getting users profile for id=%v forcing logout", userId), mlog.String("user_id", userId))
-		return nil
+func (a *App) SetIconImage(userId string, imageData *multipart.FileHeader) *model.AppError {
+	if len(*a.Config().FileSettings.DriverName) == 0 {
+		return model.NewAppError("SetIconImage", "api.user.user_icon_image.storage.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	options := a.Config().GetSanitizeOptions()
-	user.SanitizeProfile(options)
+	file, err := imageData.Open()
+	if err != nil {
+		return model.NewAppError("SetIconImage", "api.user.set_icon_image.open.app_error", nil, err.Error(), http.StatusBadRequest)
+	}
+	defer file.Close()
 
-	message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_USER_UPDATED, "", "", "", nil)
-	message.Add("user", user)
-	a.Publish(message)
+	// TODO: Sanitize SVG
+	svgInfo, err := parseSVG(file)
+	if err != nil {
+		return model.NewAppError("SetIconImage", "api.user.set_icon_image.parse.app_error", nil, err.Error(), http.StatusBadRequest)
+	}
+	if svgInfo.Width*svgInfo.Height > model.MaxImageSize {
```

Comment:
I might be mistaken, but I think the `MaxImageSize` check here is meaningless for SVGs. The check is primarily there to avoid loading a giant image into memory, but I don't think that means the same thing for SVGs?

Reply:
Yes, exactly, an svg size is not directly related to the space occupied in memory, a svg can be huge in width and height but be small in data, and all the way around.

Reply:
👍